pkg/apis/metal3/v1alpha1/baremetalhost_types.go

```diff
@@ -104,6 +104,111 @@ type BMCDetails struct {
 	CredentialsName string `json:"credentialsName"`
 }
 
+// DiskType is used to specify the disk type for a logical disk, e.g. hdd or ssd.
+type DiskType string
+
+// Valid disk types for RAID configuration
+const (
+	HDD DiskType = "hdd"
+	SSD DiskType = "ssd"
+)
+
+// ValidDiskType check whether given disk type is supported or not
+func (d DiskType) ValidDiskType () bool {
+	switch d {
+	case SSD, HDD:
+		return true
+	default:
+		return false
+	}
+}
+
+// InterfaceType is used to specify the interface for a logical disk.
+type InterfaceType string
+
+// Valid interface types for RAID configuration
+const (
+	SATA InterfaceType = "sata"
+	SCSI InterfaceType = "scsi"
+	SAS  InterfaceType = "sas"
+)
+
+// ValidInterfaceType check whether given interface type is supported or not
+func (i InterfaceType) ValidInterfaceType () bool {
+	switch i {
+	case SATA, SCSI, SAS:
+		return true
+	default:
+		return false
+	}
+}
+
+// RAIDLevel type is used to specify the RAID level for a logical disk.
+type RAIDLevel string
+
+// Valid RAID levels
+const (
+	RAID0  RAIDLevel = "0"
+	RAID1  RAIDLevel = "1"
+	RAID2  RAIDLevel = "2"
+	RAID5  RAIDLevel = "5"
+	RAID6  RAIDLevel = "6"
+	RAID10 RAIDLevel = "1+0"
+	RAID50 RAIDLevel = "5+0"
+	RAID60 RAIDLevel = "6+0"
+)
+
+// ValidRAIDLevel check whether given RAID level is supported or not
+func (l RAIDLevel) ValidRAIDLevel () bool {
+	switch l {
+	case RAID0, RAID1, RAID2, RAID5, RAID6, RAID10, RAID50, RAID60:
+		return true
+	default:
+		return false
+	}
+}
+
+// RAIDVolume defines the desired configuration of volume in RAID
+type RAIDVolume struct {
+	// Size (Integer) of the logical disk to be created in GiB.  If unspecified, "MAX" will be used.
+	SizeGb *int `json:"sizeGb,omitempty"`
+
+	// RAID level for the logical disk. The following RAID levels are supported: 0, 1, 2, 5, 6, 1+0, 5+0, 6+0.
+	RAIDLevel RAIDLevel `json:"raidLevel" required:"true"`
+
+	// Name of the volume. Should be unique within the Node. If not specified, volume name will be auto-generated.
+	VolumeName string `json:"name,omitempty"`
+
+	// Set to true if this logical disk can share physical disks with other logical disks.
+	SharePhysicalDisks *bool `json:"sharePhysicalDisks,omitempty"`
+
+	// The valid values must be "hdd" or "ssd". If this is not specified, disk type will not be a criterion to find backing physical disks
+	DiskType DiskType `json:"diskType,omitempty"`
+
+	// The valid values are "sata" or "scsi" or "sas". If this is not specified, interface type will not be a criterion to find backing physical disks.
+	InterfaceType InterfaceType `json:"interfaceType,omitempty"`
+
+	// Integer, number of disks to use for the logical disk. Defaults to minimum number of disks required
+	// for the particular RAID level.
+	NumberOfPhysicalDisks int `json:"numberOfPhysicalDisks,omitempty"`
+
+	// The name of the controller as read by the RAID interface.
+	Controller string `json:"controller,omitempty"`
```

Comment:
How is this value used?

Reply:
Yes, maybe a boolean field would simplify things.

Reply:
Thanks @dtantsur for your infomation.
I have changed to `softwareRAID` with boolean value :) However, we must define some validation due to the limitation of software RAID [1]
[1] https://docs.openstack.org/ironic/latest/admin/raid.html#software-raid